src/d/semantic/caster.d

```diff
@@ -319,6 +319,21 @@ struct Caster(bool isExplicit, alias bailoutOverride = null) {
 					return CastKind.IntToBool;
 				}
 
+				if (isFloat(bt)) {
+					// cast(float) integral|char
+					// (dmd considers char to float a valid implicit cast)
+
+					if (t == Bool) {
+						return CastKind.SignedToFloat;
```

Comment:
How come bool are considered unsigned when promoting to larger integer types, but signed when converting to floats? Are you sure this is correct? it doesn't looks like this is correct to me.

Reply:
The signedness might be wrong lemme check with ldc (dmd has no ir to copy...)

Reply:
The signed-ness was wrong, fixed.